src/utils.c

```diff
@@ -166,4 +167,24 @@ gint64 string_to_time(const char *string)
                 return 0;
 }
 
+/* see utils.h */
+gint64 time_monotonic_now(void)
+{
+        struct timespec tv_now;
+        gint64 i_now;
+
+        /* On Linux, BOOTTIME is the correct monotonic time,
+         * as BOOTTIME counts onwards during sleep. For all other
+         * POSIX compliant OSes, MONOTONIC should also count onwards
+         * during system sleep. */
+#if __linux__
+        clock_gettime(CLOCK_BOOTTIME, &tv_now);
+#else
+        clock_gettime(CLOCK_MONOTONIC, &tv_now);
+#endif
+        i_now = tv_now.tv_sec * G_USEC_PER_SEC
```

Comment:
I have a feeling this multiplication can overflow since `tv_sec` can easily be 32 bits, we should probably cast it to `gint64` first.

Reply:
Me stupid. Why does C not automatically cast to its storage value :angry: ?

Reply:
It might I'm not sure, googling around didn't help either. Still, better safe than sorry as it goes.